### DISCUSSION TITLE
Can I change the map to a backed up version from home assistant?

### DISCUSSION THREAD

**Discussion by syst3mboot** [Tasshack/dreame-vacuum#499] [Q&A] [answered]
In one area of our apartment we have tall windows, and the vacuum is constantly scanning through the windows and mapping outdoors. This means the map gets updated to something like in [this example](https://preview.redd.it/dreamebot-l-10s-ultra-keeps-looking-outside-v0-ldzkjtl97xzb1.jpg?width=640&crop=smart&auto=webp&s=9416bf1c9ac4c30cf7b87cb478b9c56f4541f8a4) after every clean. While the vacuum doesn't actually try to go outside it does increase the size of the map and ruin the layout on my dashboard.
A solution I found to this (as we cant remove sections of the map) was to create a backup of the map when it scanned it properly the first time (with the blinds closed) and simply restore the backup from the Dreametech app whenever it scans outside. This brings the map back to a reasonable size.

Now my question, is there any way we can restore the map to a backed up version through home assistant? Ideally I would like to create an automation, so after the vacuum cleans/docks, restore the map.

**Comment by Tasshack** [comment, marked as answer]
Only available on Beta.

https://github.com/Tasshack/dreame-vacuum/blob/dev/docs/map.md#backup-and-recovery
https://github.com/Tasshack/dreame-vacuum/blob/dev/docs/services.md#dreame_vacuumvacuum_restore_map

**Reply by Tasshack** [reply]
Also there is an new configuration option called `Square Map` in the Beta for rendering the map 1:1 ratio that may also solve your dashboard issue.
